File: src/main/java/com/devspace/conexfy/converters/ConAuthTypeEnumToStringWritingConverter.java
package com.devspace.conexfy.converters;

import org.springframework.core.convert.converter.Converter;
import org.springframework.data.convert.WritingConverter;
import org.springframework.lang.NonNull;

import com.devspace.conexfy.enums.ConAuthTypeEnum;

@WritingConverter
public class ConAuthTypeEnumToStringWritingConverter implements Converter<ConAuthTypeEnum, String> {

    public String convert(@NonNull ConAuthTypeEnum source) {
        return source.name();
    }
}

File: src/main/kotlin/com/devspace/conexfy/converters/ConAuthTypeEnumToStringWritingConverter.kt
package com.devspace.conexfy.converters

import com.devspace.conexfy.enums.ConAuthTypeEnum
import org.springframework.core.convert.converter.Converter
import org.springframework.data.convert.WritingConverter
import org.springframework.lang.NonNull

@WritingConverter
class ConAuthTypeEnumToStringWritingConverter : Converter<ConAuthTypeEnum, String> {
    override fun convert(@NonNull source: ConAuthTypeEnum): String {
        return source.name
    }
}